beacon_node/operation_pool/src/lib.rs

```diff
@@ -408,6 +408,9 @@ fn prune_validator_hash_map<T, F, E: EthSpec>(
 /// Compare two operation pools.
 impl<T: EthSpec + Default> PartialEq for OperationPool<T> {
     fn eq(&self, other: &Self) -> bool {
+        if self as *const _ == other as *const _ {
```

Comment:
I'm wondering if we should be using [`std::ptr::eq`](https://doc.rust-lang.org/std/ptr/fn.eq.html) as per [Why can comparing two seemingly equal pointers with == return false?](https://stackoverflow.com/questions/47489449/why-can-comparing-two-seemingly-equal-pointers-with-return-false#47490007)

@michaelsproul do you have thoughts here? I'm not particularly well versed here.

Reply:
Yeah `ptr::eq` looks like it might be cleaner. Given this `PartialEq` impl is only used in tests, I'd be interested in potentially removing it in future.

Other alternatives that I could think of are:

* Make the RHS of the `PartialEq` `&mut Self`, so that self comparisons are statically prevented (although that would be a bit gross to use)
* Use a timeout on the RwLock, knowing that it's only used for tests

Reply:
Use ptr::eq instead.